Projectoid.js

```diff
@@ -0,0 +1,292 @@
+      disableWebPreview: disableWebPreview,
+      protectContent: protectContent,
+      webhookUrl: webhookUrl,
+    },
+    onSuccess: cmd,
+  };
+  apiCall(data);
+}
+
+function forwardBroadcast(options) {
+  if (!options) {
+    throw "Error: forwardBroadcast: options not found";
+    return;
+  }
+  var from_chat_id = options.from_chat_id;
+  var message_id = options.message_id;
+
+  if (!from_chat_id || !message_id) {    
+    throw "Error: forwardBroadcast: chat id or message id was not found";
+    return;
+  }
+  var protectContent = !options.protectContent ? false : options.protectContent;
+  var webhookUrl = !options.webhookUrl ? null : options.webhookUrl;
+  var cmd = !options.command ? null : options.command;
+
+  if (!options.access_token) {
+    var access_token = Bot.getProperty("Projectoid_AccessToken");
+    if (!access_token) {
+      throw "Error: forwardBroadcast: Bot Access Token Not Found";
+      return;
+    }
+  }
+
+  let data = {
+    path: "telegram/botpanel/broadcast.php",
+    method: "post",
+    body: {
+      access_token: access_token,
+      bot_token: bot.token,
+      admin: user.telegramid,
+      method: "forwardMessage",
+      from_chat_id: from_chat_id,
+      message_id: message_id,
+      protectContent: protectContent,
+      webhookUrl: webhookUrl,
+    },
+    onSuccess: cmd,
+  };
+  apiCall(data);
+}
+
+function copyBroadcast(options) {
+  if (!options) {
+    throw "Error: copyBroadcast: options not found";
+    return;
+  }
+  var from_chat_id = options.from_chat_id;
+  var message_id = options.message_id;
+
+  if (!from_chat_id || !message_id) {
+    throw "Error: copyBroadcast: chat id or message id was not found";
+    return;
+  }
+  var protectContent = !options.protectContent ? false : options.protectContent;
+  var webhookUrl = !options.webhookUrl ? null : options.webhookUrl;
+  var cmd = !options.command ? null : options.command;
+
+  if (!options.access_token) {
+    var access_token = Bot.getProperty("Projectoid_AccessToken");
+    if (!access_token) {
+      throw "Error: copyBroadcast: Bot Access Token Not Found";
+      return;
+    }
+  }
+
+  let data = {
+    path: "telegram/botpanel/broadcast.php",
+    method: "post",
+    body: {
+      access_token: access_token,
+      bot_token: bot.token,
+      method: "copyMessage",
+      admin: user.telegramid,
+      from_chat_id: options.from_chat_id,
+      message_id: options.message_id,
+      protectContent: protectContent,
+      webhookUrl: webhookUrl,
+    },
+    onSuccess: cmd,
+  };
+  apiCall(data);
+}
+//---------------------Broadcast Functions Ends------------------------------
+
+function checkMembership(userId, chats, command) {
+  if (!userId) {
+    throw "Error: checkMembership: User id Not Found";
+    return;
+  }
+  if (!chats) {
+    throw "Error: checkMembership: Chat IDs Not Found";
+    return;
+  }
+  if (!command) {
+    throw "Error: checkMembership: command not found to return response";
```

Comment:
we need  

throw Lib_prefix + errMsg

Because we need this for error debugging